src/main/java/pl/javastart/task/TimeConverter.java

```diff
@@ -0,0 +1,120 @@
+package pl.javastart.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.zip.DataFormatException;
+
+public class TimeConverter {
+    private static final String OPERATOR_PLUS = "+";
+    private static final String YEARS = "y";
+    private static final String MONTHS = "M";
+    private static final String DAYS = "d";
+    private static final String HOURS = "h";
+    private static final String MINUTES = "m";
+    private static final String SECONDS = "s";
+
+    public LocalDateTime createDateFromString(String input) throws DataFormatException {
+        if (input.toLowerCase().startsWith("t")) {
+            return parseSpecialFormatDate(input);
+        }
+
+        List<String> dateTimeFormatterStrings = Arrays.asList("yyyy-MM-dd HH:mm:ss", "dd.MM.yyyy HH:mm:ss");
+        List<String> dateFormatterStrings = Arrays.asList("yyyy-MM-dd");
+
+        if (input.split(" ").length == 1) {
+            return parseDateTime(input, dateFormatterStrings, true);
+        } else {
+            return parseDateTime(input, dateTimeFormatterStrings, false);
+        }
+    }
+
+    private LocalDateTime parseSpecialFormatDate(String input) throws DataFormatException {
+        if (input.length() == 1) {
+            return LocalDateTime.now();
+        }
+
+        List<String> operations = new ArrayList<>(Arrays.asList(input.split("(?=[+-])")));
+        operations.remove(0); // początkowa litera "t" jest niepotrzebna dla dalszych operacji
+
+        if (!(operations.get(0).charAt(0) == '+' || operations.get(0).charAt(0) == '-')) {
+            throw new DataFormatException("Wpisano dane w nieprawidłowym formacie. Po literze \"t\" musi następować znak" +
+                    "\"+\" lub znak \"-\"");
+        }
+
+        LocalDateTime localDateTime = LocalDateTime.now();
+        for (String operation : operations) {
+            localDateTime = calculateDate(localDateTime, operation);
+        }
+        return localDateTime;
+    }
+
+    private LocalDateTime calculateDate(LocalDateTime localDateTime, String operation) {
+        String operator = operation.substring(0, 1);
+        String unit = String.valueOf(operation.charAt(operation.length() - 1));
+        int amount = Integer.parseInt(operation.substring(1, operation.length() - 1));
+        if (operator.equals(OPERATOR_PLUS)) {
+            return increaseOrDecreaseDate(localDateTime, amount, unit, true);
+        } else {
+            return increaseOrDecreaseDate(localDateTime, amount, unit, false);
+        }
```

Comment:
Tutaj dobrze sobie z tym poradziłeś, ale dorzuciłem Ci w bootcampie dodatkowy filmik o wyrażeniach regularnych, które możesz spróbować tutaj użyć. Wtedy zamiast metod substring możesz wykorzystać grupy nazwane i wydaje się to trochę bardziej eleganckie.
To poza konkursem, ale myślę, że warto spróbować